src/libraries/Common/src/Interop/OSX/Interop.libc.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class libc
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct AttrList
+        {
+            [MarshalAs(UnmanagedType.U2)] public ushort bitmapCount;
+            [MarshalAs(UnmanagedType.U2)] public ushort reserved;
+            [MarshalAs(UnmanagedType.U4)] public uint commonAttr;
+            [MarshalAs(UnmanagedType.U4)] public uint volAttr;
+            [MarshalAs(UnmanagedType.U4)] public uint dirAttr;
+            [MarshalAs(UnmanagedType.U4)] public uint fileAttr;
+            [MarshalAs(UnmanagedType.U4)] public uint forkAttr;
+
+            public const ushort ATTR_BIT_MAP_COUNT = 5;
+            public const uint ATTR_CMN_CRTIME = 0x00000200;
+        }
+
+        [DllImport(Libraries.libc, EntryPoint = "setattrlist", SetLastError = true)]
+        internal static unsafe extern int setattrlist(string path, AttrList* attrList, void* attrBuf, nint attrBufSize, uint options);
```

Comment:
`uint options` seems correct since a C# `int` is a C `long`, so a C# `uint` is a C `unsigned long`: http://www.manpagez.com/man/2/setattrlist/

@AaronRobinsonMSFT is the `string path` the correct data type to use for the first parameter of this new p/invoke?

Reply:
>uint options seems correct since a C# int is a C long, so a C# uint is a C unsigned long:

That is only true on Windows (i.e. C `long` is always 32-bits). On non-Windows the `long` is typically 32-bits on a 32-bit CPU and 64-bits on a 64-bit CPU. This is a terrible thing to discover so in .NET 6 we introduced [`CULong`](https://github.com/dotnet/runtime/pull/46401). A big thanks to @tannergooding for pushing on it.

The `string path` is correct.

Reply:
I'll change it to CULong then.

Reply:
I wonder whether some of this guidance should go into https://docs.microsoft.com/en-us/dotnet/standard/native-interop/best-practices. That's the article I usually go back to.

Reply:
@danmoseley Thanks for pointing that out: https://github.com/dotnet/docs/pull/23466